### PR TITLE
add circe to core, add codec for Gid

### PR DESCRIPTION
**Pull request by tpolecat**

### CODE REVIEW

File: modules/tests/shared/src/test/scala/lucuma/core/util/GidSuite.scala [hunk 1]
```diff
@@ -27,6 +29,7 @@ final class GidSuite extends DisciplineSuite {
   checkAll("Gid[Id]", OrderTests[Id].order)
   checkAll("Gid[Id].isoPosLong", IsoTests(Gid[Id].isoPosLong))
   checkAll("Gid[Id].fromString", PrismTests(Gid[Id].fromString))
+  checkAll("Gid[Id]", CodecTests[Id].unserializableCodec)
```
Comment (cquiroz) [inline modules/tests/shared/src/test/scala/lucuma/core/util/GidSuite.scala:32]: Why not `CodecTest[Gid[Id]]`

Comment (tpolecat) [reply modules/tests/shared/src/test/scala/lucuma/core/util/GidSuite.scala:32]: We're testing the codec for `Id`, same as `OrderTest` above.

Comment (tpolecat) [reply modules/tests/shared/src/test/scala/lucuma/core/util/GidSuite.scala:32]: So `OrderTests` and `CodecTests` are checking that the derived `Order` and `Encoder/Decoder` instances for `Id` are correct.

Comment (cquiroz) [reply modules/tests/shared/src/test/scala/lucuma/core/util/GidSuite.scala:32]: Ok, now I got it, thanks

**Comment by tpolecat** [comment]
Actually I'm following up with `GidLaws` because this is a typeclass and testing against an example isn't really the way to do it. I'm going to close this one.

**Pull request closed without merging**